README.md

```diff
@@ -1,20 +1,39 @@
 TaskCluster - Treeherder Integration
 ====================================
 
-This component reports task progress to Treeherder. For a task to be reported
-it must have the custom route `treeherder.<project>.<revisionHash>`, where
-`<project>` is the Treeherder project name and `<revisionHash>` identifies the
-Treeherder result-set that tasks should be reported to.
 
-In addition to the custom route the task must also have an extra section
-called `treeherder` with the keys: `symbol`, `groupName`, `groupSymbol` and
-`productName`. A task that should be reported to treeherder must take a form
-as follows:
+taskcluster-treeherder is a service that will respond to TaskCluster task events
+(e.g. task completed, task failed, etc) and compose a Treeherder job pulse message
+to report task status to Treeherder.
+
+# Task Configuration
+
+## Task Route
+
+For a task to be reported to treeherder, it must have the custom route in the form of:
+`<treeherder destination>.<project>.<revision>.<push/pullrequest ID>`
+
+For job messages to appear on Treeherder production, the destination of `treeherder` should
+be used.  Also, `treeherder-staging` could be used for reporting jobs to the Treeherder
+staging environment.
+
+Note: Github repos should use the `project` form of `<user>/<project>` to be recognized
+as a github source.
+
+## Treeherder job configuration
+
+All jobs that are reproted to Treeherder must have some basic information about the job
+itself, such as job symbol, job name, platform, etc.
+
+This configuration needs to be declared in the task definition under `task.extra.treeherder`
+and is validated against a [published schema](https://schemas.taskcluster.net/taskcluster-treeherder/v1/task-treeherder-config.json#).
+
+## Example  Task
 
 ```js
 {
   routes: [
-    'treeherder.<project>.<revisionHash>'
+    'treeherder.mozilla-inbound.8faa67ad6d0fda3ccf884c90acfd061d37e8558d'
```

Comment:
oh I suppose in the case of hg, a push ID will always be there, but it's not a required field (for cases where it's a github repo without a PR ID)